src/vmm/src/vmm_config/entropy.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+
+use devices::virtio::rng::{Entropy, Error as EntropyError};
+use serde::{Deserialize, Serialize};
+
+/// This struct represents the strongly typed equivalent of the json body from entropy device
+/// related requests.
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct EntropyDeviceConfig {}
+
+impl From<&Entropy> for EntropyDeviceConfig {
+    fn from(_dev: &Entropy) -> Self {
+        EntropyDeviceConfig {}
+    }
+}
+
+/// Errors that can occur while handling configuration for
+/// an entropy device
+#[derive(Debug, thiserror::Error)]
+pub enum EntropyDeviceError {
+    /// Error while create an entropy device
+    #[error("Could not create Entropy device: {0}")]
+    CreateDevice(#[from] EntropyError),
+}
+
+type Result<T> = std::result::Result<T, EntropyDeviceError>;
+
+/// A builder type used to construct an Entropy device
+#[derive(Default)]
+pub struct EntropyDeviceBuilder {
+    entropy_device: Option<Arc<Mutex<Entropy>>>,
+}
+
+impl EntropyDeviceBuilder {
+    /// Create a new instance for the builder
+    pub fn new() -> Self {
+        Self {
+            entropy_device: None,
+        }
+    }
+
+    /// Build an entropy device and return a (counted) reference to it protected by a mutex
+    pub fn build(&mut self, _config: EntropyDeviceConfig) -> Result<Arc<Mutex<Entropy>>> {
+        let dev = Arc::new(Mutex::new(Entropy::new()?));
+        self.entropy_device = Some(dev.clone());
+
+        Ok(dev)
+    }
+
+    /// Insert a new entropy device from a configuration object
+    pub fn insert(&mut self, config: EntropyDeviceConfig) -> Result<()> {
+        let _ = self.build(config)?;
+        Ok(())
+    }
+
+    /// Get a reference to the entropy device, if present
+    pub fn get(&self) -> Option<&Arc<Mutex<Entropy>>> {
+        self.entropy_device.as_ref()
+    }
+
+    /// Get the configuration of the entropy device (if any)
+    pub fn config(&self) -> Option<EntropyDeviceConfig> {
+        self.entropy_device
+            .as_ref()
+            .map(|dev| EntropyDeviceConfig::from(dev.lock().unwrap().deref()))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    pub(crate) fn default_config() -> EntropyDeviceConfig {
+        EntropyDeviceConfig {}
+    }
+
+    #[test]
+    fn test_entropy_device_create() {
+        let config = default_config();
+        let mut builder = EntropyDeviceBuilder::new();
+        assert!(builder.get().is_none());
+
+        builder.insert(config.clone()).unwrap();
+        assert!(builder.get().is_some());
```

Comment:
Nit: Prefer I tend to prefer `assert_eq!` here.

Reply:
This would require to `Entropy` to be `PartialEq`, right? If that's the case, I don't think it's doable. There are various fields that are not `PartialEq`.